testsuite/gcmc_tests.py

```diff
@@ -0,0 +1,45 @@
+# Import pyMBE and other libraries
+import pyMBE
+from lib import analysis
+import os
+import tempfile
+import subprocess
+import numpy as np
+import pandas as pd
+import argparse
+
+# Create an instance of pyMBE library
+pmb = pyMBE.pymbe_library()
+
+# Command line arguments
+parser = argparse.ArgumentParser(description='Test of the GCMC implementation in pyMBE.')
+parser.add_argument('--mode',
+                    type=str,
+                    default= "ideal",
+                    help='Set if an ideal or interacting system is simulated.')
+args = parser.parse_args()
+
+script_path=pmb.get_resource(f"samples/salt_solution_gcmc.py")
+salt_concentrations=[0.0001, 0.001, 0.01, 0.1]
+
+rtol=0.05 # relative tolerance
+atol=0.0 # absolute tolerance
+
+if args.mode == "ideal":
+    print(f"*** Running test for GCMC of salt solution (ideal). ***")
+elif args.mode == "interacting":
+    print(f"*** Running test for GCMC of salt solution (interacting). ***")
+with tempfile.TemporaryDirectory() as time_series_path:
+    for c_salt_res in salt_concentrations:
+        print(f"c_salt_res = {c_salt_res}")
+        run_command=["python3", script_path, "--c_salt_res", str(c_salt_res), "--mode", "ideal", "--output", time_series_path, "--mode", args.mode, "--no_verbose"]
+        print(subprocess.list2cmdline(run_command))
+        subprocess.check_output(run_command)
+    # Analyze all time series
+    data=analysis.analyze_time_series(path_to_datafolder=time_series_path)
```

Comment:
@davidbbeyer I would prefer to change the logic of the test a bit to avoid having to call the script twice:
1) Encapsulate this lines into a function (for example `test_gcmc`)
2) remove the argparse input
3) simply call the function twice in the script for the interacting and the ideal case.
The reason is that then in the Makefile each line correspond to call to a different testing block, preventing it to grow too much in the future when we add more tests.

Otherwise this PR looks very good to me.

Reply:
@pm-blanco I have adjusted the test accordingly